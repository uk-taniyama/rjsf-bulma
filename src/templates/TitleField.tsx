import type { FC } from "react";
import type { TitleFieldProps } from "@rjsf/utils";
import { Required } from "../ui";

const TitleField: FC<TitleFieldProps> = ({ id, title, required, uiSchema }) => (
  <div id={id} className="title-field is-size-6">
    {(uiSchema && uiSchema["ui:title"]) || title}
    <Required required={required} />
  </div>
);

export default TitleField;
