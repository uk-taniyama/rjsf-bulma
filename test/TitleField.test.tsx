import React from "react";
import renderer from "react-test-renderer";

import TitleField from "../src/templates/TitleField";
import { mockRegistry } from "./helpers/createMocks";

describe("TitleField", () => {
  test("simple", () => {
    const registry = mockRegistry();
    const tree = renderer
      .create(<TitleField id="testid" title="Hello" registry={registry} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
