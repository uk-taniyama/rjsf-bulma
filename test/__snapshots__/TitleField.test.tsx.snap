// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleField simple 1`] = `
<div
  className="title-field"
  id="testid"
>
  Hello
</div>
`;
